src/js/common/schemaform/SchemaForm.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import _ from 'lodash/fp';
+import Form from 'react-jsonschema-form';
+
+import { uiSchemaValidate, transformErrors } from './validation';
+import Address from './Address';
+import FieldTemplate from './FieldTemplate';
+import * as reviewWidgets from './review/widgets';
+import ReviewFieldTemplate from './review/ReviewFieldTemplate';
+import widgets from './widgets/index';
+import ObjectField from './ObjectField';
+import ArrayField from './ArrayField';
+import ReviewObjectField from './review/ObjectField';
+import { scrollToFirstError } from '../utils/helpers';
+
+const fields = {
+  ObjectField,
+  ArrayField,
+  address: Address
+};
+
+const reviewFields = {
+  ObjectField: ReviewObjectField,
+  ArrayField,
+  address: ReviewObjectField
+};
+
+/*
+ * Each page uses this component and passes in config. This is where most of the page level
+ * form logic should live.
+ */
+class SchemaForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.validate = this.validate.bind(this);
```

Comment:
Do you mean the auto binding you were using the other day for class methods?

```
validate = () => {
...
}
```

I don't have a strong preference either way. I didn't know about that syntax until I saw it a few weeks ago in the react-jsonschema-form code. I'm torn about it because it might reduce bugs caused by forgetting to bind in the constructor, but the syntax itself seems non-obvious and hard to google.